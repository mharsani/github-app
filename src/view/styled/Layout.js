import styled from 'styled-components';
import { Image } from 'react-bootstrap';

const Wrapper = styled.div`
  height: 100%;
`;

const Row = styled.div`
  padding: 3.75em 0;
`;

const ImageUser = styled(Image)`
  padding: 1em 0;
  max-width: 20rem;
`;


export { Wrapper, Row, ImageUser };
