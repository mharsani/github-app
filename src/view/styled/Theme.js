import { createGlobalStyle } from 'styled-components';

const theme = {
  breakpoints: [30, 36, 48, 62, 75, 100],
  fontSizes: [0.75, 0.875, 1, 1.125, 1.5, 1.875],
  colors: {
    color01: '#FFF',
    color02: '#000',
    color03: 'rgba(0,0,0,.125)',
    color04: '#ffffff80'
  },
  fontWeights: {},
  lineHeights: {},
  letterSpacings: {},
  shadows: {},
  borders: {},
  radii: {},
  opacity: {}
};

const GlobalStyle = createGlobalStyle`
  html,
  body {
    font-size: 16px;
    background-color: #fff;
  }
`;

export { theme, GlobalStyle };
