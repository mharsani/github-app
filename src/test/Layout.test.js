import React from "react";
import renderer from "react-test-renderer";

import { theme } from "../view/styled/Theme";
import {
  Wrapper,
  Row,
  ImageUser,
  CardUser,
  CardContainer,
  Overlay,
  SpinStyle
} from "../view/styled/Layout";

describe("Layout Component", () => {
  test("Wrapper styled component test", () => {
    const tree = renderer.create(<Wrapper theme={theme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Row styled component test", () => {
    const tree = renderer.create(<Row theme={theme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("CardUser styled component test", () => {
    const tree = renderer.create(<CardUser theme={theme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("ImageUser styled component test reverse", () => {
    const tree = renderer.create(<ImageUser theme={theme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("CardContainer styled component test ", () => {
    const tree = renderer.create(<CardContainer theme={theme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Overlay styled component test ", () => {
    const tree = renderer.create(<Overlay theme={theme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("SpinStyle styled component", () => {
    const tree = renderer.create(<SpinStyle theme={theme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
