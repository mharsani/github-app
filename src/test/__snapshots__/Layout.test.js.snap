// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component CardContainer styled component test  1`] = `
<div
  className="sc-ifAKCX qDbXr"
/>
`;

exports[`Layout Component CardUser styled component test 1`] = `
<div
  className="sc-bxivhb jgNNFH"
/>
`;

exports[`Layout Component CardUser styled component test reverse 1`] = `
<div
  className="sc-bxivhb jgNNFH"
/>
`;

exports[`Layout Component Column styled component test dont reverse, table and small 1`] = `
<img
  className="sc-htpNat gqTyeo"
  theme={
    Object {
      "borders": Object {},
      "breakpoints": Array [
        30,
        36,
        48,
        62,
        75,
        100,
      ],
      "colors": Object {
        "color01": "#FFF",
        "color02": "#000",
        "color03": "rgba(0,0,0,.125)",
        "color04": "#ffffff80",
      },
      "fontSizes": Array [
        0.75,
        0.875,
        1,
        1.125,
        1.5,
        1.875,
      ],
      "fontWeights": Object {},
      "letterSpacings": Object {},
      "lineHeights": Object {},
      "opacity": Object {},
      "radii": Object {},
      "shadows": Object {},
    }
  }
/>
`;

exports[`Layout Component Column styled component test reverse 1`] = `
<div
  className="sc-bxivhb jgNNFH"
/>
`;

exports[`Layout Component Column styled component test small 1`] = `
<div
  className="sc-EHOje exEyFl"
/>
`;

exports[`Layout Component Column styled component test small 2`] = `
<div
  className="sc-bZQynM hOiolA"
/>
`;

exports[`Layout Component Column styled component test table 1`] = `
<div
  className="sc-ifAKCX qDbXr"
/>
`;

exports[`Layout Component ImageUser styled component test reverse 1`] = `
<img
  className="sc-htpNat gqTyeo"
  theme={
    Object {
      "borders": Object {},
      "breakpoints": Array [
        30,
        36,
        48,
        62,
        75,
        100,
      ],
      "colors": Object {
        "color01": "#FFF",
        "color02": "#000",
        "color03": "rgba(0,0,0,.125)",
        "color04": "#ffffff80",
      },
      "fontSizes": Array [
        0.75,
        0.875,
        1,
        1.125,
        1.5,
        1.875,
      ],
      "fontWeights": Object {},
      "letterSpacings": Object {},
      "lineHeights": Object {},
      "opacity": Object {},
      "radii": Object {},
      "shadows": Object {},
    }
  }
/>
`;

exports[`Layout Component Overlay styled component test  1`] = `
<div
  className="sc-EHOje exEyFl"
/>
`;

exports[`Layout Component Row styled component test 1`] = `
<div
  className="sc-bwzfXH gIdJCH"
/>
`;

exports[`Layout Component SpinStyle styled component 1`] = `
<div
  className="sc-bZQynM hOiolA"
/>
`;

exports[`Layout Component Wrapper styled component test 1`] = `
<div
  className="sc-bdVaJa dGIfxd"
/>
`;
